# **Introduction**

# **Dependencies**

In [124]:
import pandas as pd # csv file manipulation
import numpy as np
import plotly.express as px # data visualizatoin library

In [125]:
life_expect = pd.read_csv("/kaggle/input/life-expectency/Life Expectancy Data_HV22.csv")
life_expect

Country  Year      Status  Life expectancy   Adult Mortality  \
0     Afghanistan  2015  Developing              65.0            263.0   
1     Afghanistan  2014  Developing              59.9            271.0   
2     Afghanistan  2013  Developing              59.9            268.0   
3     Afghanistan  2012  Developing              59.5            272.0   
4     Afghanistan  2011  Developing              59.2            275.0   
...           ...   ...         ...               ...              ...   
2933     Zimbabwe  2004  Developing              44.3            723.0   
2934     Zimbabwe  2003  Developing              44.5            715.0   
2935     Zimbabwe  2002  Developing              44.8             73.0   
2936     Zimbabwe  2001  Developing              45.3            686.0   
2937     Zimbabwe  2000  Developing              46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   \
0                62     0.01               71.279624         65.0      1154   
1                64     0.01               73.523582         62.0       492   
2                66     0.01               73.219243         64.0       430   
3                69     0.01               78.184215         67.0      2787   
4                71     0.01                7.097109         68.0      3013   
...             ...      ...                     ...          ...       ...   
2933             27     4.36                0.000000         68.0        31   
2934             26     4.06                0.000000          7.0       998   
2935             25     4.43                0.000000         73.0       304   
2936             25     1.72                0.000000         76.0       529   
2937             24     1.68                0.000000         79.0      1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
0     ...    6.0               8.16         65.0        0.1  584.259210   
1     ...   58.0               8.18         62.0        0.1  612.696514   
2     ...   62.0               8.13         64.0        0.1  631.744976   
3     ...   67.0               8.52         67.0        0.1  669.959000   
4     ...   68.0               7.87         68.0        0.1   63.537231   
...   ...    ...                ...          ...        ...         ...   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358878   

      Population   thinness  1-19 years   thinness 5-9 years  \
0     33736494.0                   17.2                 17.3   
1       327582.0                   17.5                 17.5   
2     31731688.0                   17.7                 17.7   
3      3696958.0                   17.9                 18.0   
4      2978599.0                   18.2                 18.2   
...          ...                    ...                  ...   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
0                               0.479       10.1  
1                               0.476       10.0  
2                               0.470        9.9  
3                               0.463        9.8  
4                               0.454        9.5  
...                               ...        ...  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427      

*A description of all the columns *


Country: Country

Year: Year

Status: Developed or Developing status

Life expectancy: Life Expectancy in age

Adult Mortality: Adult Mortality Rates of both sexes (probability of dying between 15 and 60 years per 1000 population)

infant deaths: Number of Infant Deaths per 1000 population

Alcohol: Alcohol, recorded per capita (15+) consumption (in litres of pure alcohol)

percentage expenditure:Expenditure on health as a percentage of Gross Domestic Product per capita(%)

Hepatitis B: Hepatitis B (HepB) immunization coverage among 1-year-olds (%)

Measles: Measles - number of reported cases per 1000 population

BMI: Average Body Mass Index of entire population

under-five deaths: Number of under-five deaths per 1000 population

Polio: Polio (Pol3) immunization coverage among 1-year-olds (%)

Total expenditure: General government expenditure on health as a percentage of total government expenditure (%)

Diphtheria: Diphtheria tetanus toxoid and pertussis (DTP3) immunization coverage among 1-year-olds (%)

HIV/AIDS: Deaths per 1 000 live births HIV/AIDS (0-4 years)

GDP: Gross Domestic Product per capita (in USD)

Population: Population of the country

thinness 1-19 years: Prevalence of thinness among children and adolescents for Age 10 to 19 (%)

thinness 5-9 years: Prevalence of thinness among children for Age 5 to 9(%)

Income composition of resources: Human Development Index in terms of income composition of resources (index ranging from 0 to 1)

Schooling: Number of years of Schooling(years)

# **Preprocessing**

In [126]:
life_expect.info()# to find info about the columns and missing values

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2938 entries, 0 to 2937
Data columns (total 22 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Country                          2938 non-null   object 
 1   Year                             2938 non-null   int64  
 2   Status                           2938 non-null   object 
 3   Life expectancy                  2928 non-null   float64
 4   Adult Mortality                  2928 non-null   float64
 5   infant deaths                    2938 non-null   int64  
 6   Alcohol                          2744 non-null   float64
 7   percentage expenditure           2938 non-null   float64
 8   Hepatitis B                      2385 non-null   float64
 9   Measles                          2938 non-null   int64  
 10   BMI                             2904 non-null   float64
 11  under-five deaths                2938 non-null   int64  
 12  Polio               

In [127]:
life_expect.describe()

Year  Life expectancy   Adult Mortality  infant deaths  \
count  2938.000000       2928.000000      2928.000000    2938.000000   
mean   2007.518720         69.224932       164.796448      30.303948   
std       4.613841          9.523867       124.292079     117.926501   
min    2000.000000         36.300000         1.000000       0.000000   
25%    2004.000000         63.100000        74.000000       0.000000   
50%    2008.000000         72.100000       144.000000       3.000000   
75%    2012.000000         75.700000       228.000000      22.000000   
max    2015.000000         89.000000       723.000000    1800.000000   

           Alcohol  percentage expenditure  Hepatitis B       Measles   \
count  2744.000000             2938.000000  2385.000000    2938.000000   
mean      4.602861              738.251295    80.940461    2419.592240   
std       4.052413             1987.914858    25.070016   11467.272489   
min       0.010000                0.000000     1.000000       0.000000   
25%       0.877500                4.685343    77.000000       0.000000   
50%       3.755000               64.912906    92.000000      17.000000   
75%       7.702500              441.534144    97.000000     360.250000   
max      17.870000            19479.911610    99.000000  212183.000000   

              BMI   under-five deaths         Polio  Total expenditure  \
count  2904.000000         2938.000000  2919.000000         2712.00000   
mean     38.321247           42.035739    82.550188            5.93819   
std      20.044034          160.445548    23.428046            2.49832   
min       1.000000            0.000000     3.000000            0.37000   
25%      19.300000            0.000000    78.000000            4.26000   
50%      43.500000            4.000000    93.000000            5.75500   
75%      56.200000           28.000000    97.000000            7.49250   
max      87.300000         2500.000000    99.000000           17.60000   

       Diphtheria      HIV/AIDS            GDP    Population  \
count  2919.000000  2938.000000    2490.000000  2.286000e+03   
mean     82.324084     1.742103    7483.158469  1.275338e+07   
std      23.716912     5.077785   14270.169342  6.101210e+07   
min       2.000000     0.100000       1.681350  3.400000e+01   
25%      78.000000     0.100000     463.935626  1.957932e+05   
50%      93.000000     0.100000    1766.947595  1.386542e+06   
75%      97.000000     0.800000    5910.806335  7.420359e+06   
max      99.000000    50.600000  119172.741800  1.293859e+09   

        thinness  1-19 years   thinness 5-9 years  \
count            2904.000000          2904.000000   
mean                4.839704             4.870317   
std                 4.420195             4.508882   
min                 0.100000             0.100000   
25%                 1.600000             1.500000   
50%                 3.300000             3.300000   
75%                 7.200000             7.200000   
max                27.700000            28.600000   

       Income composition of resources    Schooling  
count                      2771.000000  2775.000000  
mean                          0.627551    11.992793  
std                           0.210904     3.358920  
min                           0.000000     0.000000  
25%                           0.493000    10.100000  
50%                           0.677000    12.300000  
75%                           0.779000    14.300000  
max                           0.948000    20.700000

Since many columns have missing rows we will start preprocessing

In [128]:
life_expect.isna().sum()# Check how many rows are missing in each column

Country                              0
Year                                 0
Status                               0
Life expectancy                     10
Adult Mortality                     10
infant deaths                        0
Alcohol                            194
percentage expenditure               0
Hepatitis B                        553
Measles                              0
 BMI                                34
under-five deaths                    0
Polio                               19
Total expenditure                  226
Diphtheria                          19
 HIV/AIDS                            0
GDP                                448
Population                         652
 thinness  1-19 years               34
 thinness 5-9 years                 34
Income composition of resources    167
Schooling                          163
dtype: int64

In [129]:
life_expect = life_expect.rename(columns={
    'Life expectancy ':'Life_Expectancy',
    'Adult Mortality': 'Adult_Mortality',
    'infant deaths':'Infant_Deaths',
    'percentage expenditure':'Percentage_Expenditure',
    'Hepatitis B': 'Hepatitis_B',
    'Measles ':'Measles',
    ' BMI ':'BMI',
    'under-five deaths ':'Under-five_Deaths',
    'Total expenditure':'Total_Expenditure',
    'Diphtheria ':'Diphtheria',
    ' HIV/AIDS':'HIV/AIDS',
    ' thinness  1-19 years':'Thinness_1-19_Years',
    ' thinness 5-9 years':'Thinness_5-9_Years',
    'Income composition of resources':'Income_Composition_of_Resources'
})

The column name have unnecessary spaces and wrong capitalization. We will rename these columns for increased readability

We will separate the columns with more than a hundred missing values and those with less than a hundred missing values

In [130]:
missing_rows_columns_low = ["Life_Expectancy" , "Adult_Mortality" , "BMI" , "Under-five_Deaths" , "Polio", "Diphtheria", "Thinness_1-19_Years", "Thinness_5-9_Years"]
missing_rows_columns_high = ["Alcohol" , "Hepatitis_B" , "Total_Expenditure" , "GDP" , "Population" , "Income_Composition_of_Resources" , "Schooling"]
print("Columns with less than 100 missing values")
display(life_expect[missing_rows_columns_low])
print("Columns with more than 100 missing values")
display(life_expect[missing_rows_columns_high])

Columns with less than 100 missing values


Life_Expectancy  Adult_Mortality   BMI  Under-five_Deaths  Polio  \
0                65.0            263.0  19.1                 83    6.0   
1                59.9            271.0  18.6                 86   58.0   
2                59.9            268.0  18.1                 89   62.0   
3                59.5            272.0  17.6                 93   67.0   
4                59.2            275.0  17.2                 97   68.0   
...               ...              ...   ...                ...    ...   
2933             44.3            723.0  27.1                 42   67.0   
2934             44.5            715.0  26.7                 41    7.0   
2935             44.8             73.0  26.3                 40   73.0   
2936             45.3            686.0  25.9                 39   76.0   
2937             46.0            665.0  25.5                 39   78.0   

      Diphtheria  Thinness_1-19_Years  Thinness_5-9_Years  
0           65.0                 17.2                17.3  
1           62.0                 17.5                17.5  
2           64.0                 17.7                17.7  
3           67.0                 17.9                18.0  
4           68.0                 18.2                18.2  
...          ...                  ...                 ...  
2933        65.0                  9.4                 9.4  
2934        68.0                  9.8                 9.9  
2935        71.0                  1.2                 1.3  
2936        75.0                  1.6                 1.7  
2937        78.0                 11.0                11.2  

[2938 rows x 8 columns]

Columns with more than 100 missing values


Alcohol  Hepatitis_B  Total_Expenditure         GDP  Population  \
0        0.01         65.0               8.16  584.259210  33736494.0   
1        0.01         62.0               8.18  612.696514    327582.0   
2        0.01         64.0               8.13  631.744976  31731688.0   
3        0.01         67.0               8.52  669.959000   3696958.0   
4        0.01         68.0               7.87   63.537231   2978599.0   
...       ...          ...                ...         ...         ...   
2933     4.36         68.0               7.13  454.366654  12777511.0   
2934     4.06          7.0               6.52  453.351155  12633897.0   
2935     4.43         73.0               6.53   57.348340    125525.0   
2936     1.72         76.0               6.16  548.587312  12366165.0   
2937     1.68         79.0               7.10  547.358878  12222251.0   

      Income_Composition_of_Resources  Schooling  
0                               0.479       10.1  
1                               0.476       10.0  
2                               0.470        9.9  
3                               0.463        9.8  
4                               0.454        9.5  
...                               ...        ...  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427       10.0  
2936                            0.427        9.8  
2937                            0.434        9.8  

[2938 rows x 7 columns]

# **Column Analysis**

### Life Expectency

We will check the life expectency column. 

Since many of our prections are based on life expectency we will drop the missing values in data. The number of missing rows in life_expectency column is 10.

In [131]:
life_expect = life_expect.dropna(subset=['Life_Expectancy'])

In [132]:
fig = px.histogram(life_expect, x="Life_Expectancy")
fig.show()

the maximum life expectency is 89 according to the dataset <br>
the minimum life expectency is 36.3 according to the dataset<br>
the standard deviation of life expectency is 9.52

In [133]:
def numerical_column_analysis(col):
    print("The maximum {} is {} according to the dataset".format(col , life_expect[col].max()))
    print("The minimum {} is {} according to the dataset".format(col , life_expect[col].min()))
    print("the standard deviation of {} is {}".format(col , life_expect[col].std()))
    print("the median {} is {}".format(col , life_expect[col].median()))
    print("the mean of {} is {}".format(col , life_expect[col].mean()))
numerical_column_analysis("Life_Expectancy")

The maximum Life_Expectancy is 89.0 according to the dataset
The minimum Life_Expectancy is 36.3 according to the dataset
the standard deviation of Life_Expectancy is 9.523867487824301
the median Life_Expectancy is 72.1
the mean of Life_Expectancy is 69.22493169398908


the mean life expectency is 69.22 while the median is 72.1 which means the column is left-skewed

### Adult Mortality and Infant Mortality

In [134]:
from plotly.subplots import make_subplots
import plotly.graph_objects as go

fig = make_subplots(
    rows=2, cols=1,
    subplot_titles=("Adult_Mortality","Infant_Deaths"))
# fig = go.Figure(data=[go.Histogram(x=x)])
fig.append_trace(go.Histogram(x=life_expect["Adult_Mortality"]), row=1, col=1)
fig.append_trace(go.Histogram(x=life_expect["Infant_Deaths"]), row=2, col=1)
# fig.append_trace(go.Figure(life_expect=[go.Histogram(x="Infant_Deaths")]), row=2, col=1)

fig.show()


In [135]:
print("Adult_morality")
numerical_column_analysis("Adult_Mortality")
print("Infant_Deaths")
numerical_column_analysis("Infant_Deaths")

Adult_morality
The maximum Adult_Mortality is 723.0 according to the dataset
The minimum Adult_Mortality is 1.0 according to the dataset
the standard deviation of Adult_Mortality is 124.29207900342186
the median Adult_Mortality is 144.0
the mean of Adult_Mortality is 164.79644808743168
Infant_Deaths
The maximum Infant_Deaths is 1800 according to the dataset
The minimum Infant_Deaths is 0 according to the dataset
the standard deviation of Infant_Deaths is 118.11444959808102
the median Infant_Deaths is 3.0
the mean of Infant_Deaths is 30.407445355191257


Adult Mortality is right skewed while infant mortality is left skewed. The 10 missing values in Adult mortality were also the missing values in life  expectency which have been taken care of. There seems to be a fault in the data since infant mortality is out of a thousand however the maximum value is 1800

In [140]:
life_expect[life_expect['Infant_Deaths'] > 1000]

Country  Year      Status  Life_Expectancy  Adult_Mortality  \
1189   India  2012  Developing             67.3             19.0   
1190   India  2011  Developing             66.8            193.0   
1191   India  2010  Developing             66.4            196.0   
1192   India  2009  Developing             66.0              2.0   
1193   India  2008  Developing             65.5             23.0   
1194   India  2007  Developing             65.2             26.0   
1195   India  2006  Developing             64.8             28.0   
1196   India  2005  Developing             64.4            211.0   
1197   India  2004  Developing             64.0            214.0   
1198   India  2003  Developing             63.7            216.0   
1199   India  2002  Developing             63.3            219.0   
1200   India  2001  Developing             62.9            222.0   
1201   India  2000  Developing             62.5            224.0   

      Infant_Deaths  Alcohol  Percentage_Expenditure  Hepatitis_B  Measles  \
1189           1100     3.10               64.969645         73.0    18668   
1190           1100     3.00               64.605901         44.0    33634   
1191           1200     2.77               57.733599         38.0    31458   
1192           1300     2.50                0.844186         37.0    56188   
1193           1300     1.93               43.030433         29.0    44258   
1194           1400     1.59                5.234770          6.0    41144   
1195           1500     1.37               34.859427          6.0    64185   
1196           1500     1.27                3.509637          8.0    36711   
1197           1600     1.20               27.338009          6.0    55443   
1198           1700     1.19               19.480868          NaN    47147   
1199           1700     1.10               17.812056          NaN    40044   
1200           1800     1.00               19.003406          NaN    51780   
1201           1800     0.93               19.266157          NaN    38835   

      ...  Polio  Total_Expenditure  Diphtheria  HIV/AIDS          GDP  \
1189  ...   79.0               4.39        82.0       0.2  1446.985410   
1190  ...   79.0               4.33        82.0       0.2  1461.671957   
1191  ...   76.0               4.28        79.0       0.2  1345.771530   
1192  ...   73.0               4.38        74.0       0.2    19.317765   
1193  ...   69.0               4.34         7.0       0.3   991.484634   
1194  ...   67.0               4.23        64.0       0.3   118.166373   
1195  ...   66.0               4.25        65.0       0.3   792.259698   
1196  ...   65.0               4.28        65.0       0.3    77.819000   
1197  ...   58.0               4.22        63.0       0.3   621.318377   
1198  ...   57.0               4.30        61.0       0.3   541.135228   
1199  ...   58.0               4.40        59.0       0.3   466.284200   
1200  ...   58.0               4.50        59.0       0.3   447.138966   
1201  ...   57.0               4.26        58.0       0.3   438.864634   

        Population  Thinness_1-19_Years  Thinness_5-9_Years  \
1189  1.263659e+08                 26.9                27.6   
1190  1.247236e+08                 26.9                27.7   
1191  1.239869e+07                 27.0                27.8   
1192  1.214271e+08                 27.0                27.8   
1193  1.197147e+08                 27.0                27.9   
1194  1.179681e+09                 27.1                28.0   
1195  1.161978e+09                 27.1                28.0   
1196  1.144119e+09                 27.2                28.1   
1197  1.126136e+09                 27.2                28.2   
1198  1.182785e+07                 27.3                28.3   
1199  1.898711e+07                 27.4                28.4   
1200  1.714779e+08                 27.5                28.5   
1201  1.535912e+06                 27.7                28.6   

      Income_Composition_of_Resources  

In [146]:
life_expect[life_expect['Year'] > 2012][life_expect['Country'] == "India"]

/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:1: UserWarning:

Boolean Series key will be reindexed to match DataFrame index.



Country  Year      Status  Life_Expectancy  Adult_Mortality  \
1186   India  2015  Developing             68.3            181.0   
1187   India  2014  Developing             68.0            184.0   
1188   India  2013  Developing             67.6            187.0   

      Infant_Deaths  Alcohol  Percentage_Expenditure  Hepatitis_B  Measles  \
1186            910      NaN                0.000000         87.0    90387   
1187            957     3.07               86.521539         79.0    79563   
1188           1000     3.11               67.672304          7.0    13822   

      ...  Polio  Total_Expenditure  Diphtheria  HIV/AIDS          GDP  \
1186  ...   86.0                NaN        87.0       0.2  1613.188780   
1187  ...   84.0               4.69        85.0       0.2  1573.118890   
1188  ...   82.0               4.53        83.0       0.2  1452.195373   

        Population  Thinness_1-19_Years  Thinness_5-9_Years  \
1186  1.395398e+06                 26.7                27.3   
1187  1.293859e+09                 26.8                27.4   
1188  1.278562e+08                 26.8                27.5   

      Income_Composition_of_Resources  Schooling  
1186                            0.615       11.6  
1187                            0.607       11.6  
1188                            0.599       11.5  

[3 rows x 22 columns]